Lib/test/test_unittest/testmock/testthreadingmock.py

```diff
@@ -197,5 +198,24 @@ def test_reset_mock_resets_wait(self):
         m.assert_called_once()
 
 
+class TestRaceConditions(unittest.TestCase):
+    def test_str(self):
+        def f():
+            for m in mocks:
+                try:
+                    str(m)
+                except TypeError:
+                    nonlocal fail
+                    fail = True
+        fail = False
+        mocks = [MagicMock() for _ in range(1000)]
+        threads = [Thread(target=f) for _ in range(10)]
+        for thread in threads:
+            thread.start()
+        for thread in threads:
+            thread.join()
+        self.assertFalse(fail)
```

Comment:
How long does this test take to run? Does it _always _ fail before the change in this PR?

Reply:
Thanks for the review. Great point. I've rewritten the test with 0.2 seconds of deliberate delays (have to patch `MagicMock.__setattr__` to make it happen) so that the execution order can be guaranteed to trigger a race condition before the PR and pass after the PR.